src/blob_gc_picker.cc

```diff
@@ -71,8 +67,17 @@ std::unique_ptr<BlobGC> BasicBlobGCPicker::PickBlobGC(
                   "got batch size %" PRIu64 ", estimate output %" PRIu64
                   " bytes",
                   batch_size, estimate_output_size);
-  if (blob_files.empty() || batch_size < cf_options_.min_gc_batch_size)
+  if (blob_files.empty() || batch_size < cf_options_.min_gc_batch_size) {
+    return nullptr;
+  }
+  // if there is only one small file to merge, no need to perform
+  if (blob_files.size() == 1 &&
```

Comment:
I think we shouldn't merge small files if the end result is still less than `merge_small_file_threshold`, but let's do it next time.